### PR TITLE
Configurable headers and parameters for Group Service Api Calls.

### PR DESCRIPTION
**Pull request by rbkasat**
Updated the Group Service Configurations to support static headers and parameters to be configured.
Some API's need basic auth, some need Hmac token as request param, which can be statically configured in the configuration Yml. New updates won't affect the existing configurations and won't take effect unless configured.

### CODE REVIEW

**Comment by sscarduzio** [comment]
Awesome! Thanks @rbkasat.
I have a really simple question: why is it called "default_query_parameters" and "default_headers"? I mean, why "default"?

File: core/src/test/java/tech/beshu/ror/acl/blocks/rules/impl/GroupsProviderAuthorizationAsyncRuleTests.java [hunk 1]
```diff
@@ -59,7 +60,7 @@ public void testUserAuthorizationSuccess() throws Exception {
     result = createRuleRunMatch(Lists.newArrayList("group1", "group3"),
         "/complex/groups", "POST",
         ImmutableMap.of("Content-Type", "application/x-www-form-urlencoded",
-            "Authorization", "Basic base64_encoded_token"),
```
Comment (sscarduzio) [inline core/src/test/java/tech/beshu/ror/acl/blocks/rules/impl/GroupsProviderAuthorizationAsyncRuleTests.java]: Why not to use the standard auth header "Authorization"?

**Comment by rbkasat** [comment]
Since these headers are static and will be added to each request by default. 

Just wanted to make expectation clear that the headers specified in this block will not be scraped from a request eg: if ES API call comes with a bunch of headers, people may want to use those headers instead of statically configured ones from yml file. 
same for request parameters.

This would solve the purpose mainly for authentication while calling APIs. Authentication details are generally static.

if a default doesn't sound very good, please suggest, I can get rid of it.

**Comment by sscarduzio** [comment]
It's not that it doesn't sound good, it would be nice to have the headers and QP values being replaced by [@{dynamic_variables}](https://github.com/beshu-tech/readonlyrest-docs/blob/master/elasticsearch.md#dynamic-variables) like we have for the [kibana_index rule](https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/blob/master/core/src/main/java/tech/beshu/ror/acl/blocks/rules/impl/KibanaIndexSyncRule.java#L55), and many others:



This will enable the reuse of the values of the incoming HTTP request headers, or the resolved user's name. What do you think?

File: core/src/test/java/tech/beshu/ror/acl/blocks/rules/impl/GroupsProviderAuthorizationAsyncRuleTests.java (hunk 1 above)
Comment (rbkasat) [reply core/src/test/java/tech/beshu/ror/acl/blocks/rules/impl/GroupsProviderAuthorizationAsyncRuleTests.java]: this is just a test to show even basic auth can be simply passed in one argument, no extra logic required. the header can be anything. it's just a name-value pair for users.

**Comment by rbkasat** [comment]
yes, sounds like a good idea but can be done in iterations :) if you agree. the first iteration static params and second iteration may be dynamic params unless its a minor change?

**Pull request merged**